lua/metals.lua

```diff
@@ -227,4 +227,15 @@ M['metals/publishDecorations'] = function(err, _, decorations)
   end
 end
 
+-- Notify the server when document has been focused
+-- This needs to be called in the appropriate autocommand, i.e. FocusGained
+M.did_focus = function()
+  local focused_uri = vim.uri_from_bufnr(0)
+  vim.lsp.buf_request(0, 'metals/didFocusTextDocument', focused_uri, function(err, _, _)
```

Comment:
Are you ready for this??

This is sending the `didFocusTextDocument` *request* to the server, and this needs to be a notify. Changing this to:

```suggestion
  vim.lsp.buf_notify(0, 'metals/didFocusTextDocument', focused_uri, function(err, _, _)
```
fixes the issue.

Reply:
Wow! TIL about notify, thank you for digging into that, I was sure it had to do with the interaction, but I didn't know there was another pattern.

Reply:
Give me bigger :man_facepalming: